mslib/msui/flighttrack.py

```diff
@@ -644,13 +645,19 @@ def load_from_ftml(self, filename):
         _dirname, _name = os.path.split(filename)
         _fs = fs.open_fs(_dirname)
         xml_content = _fs.readtext(_name)
-        name = os.path.basename(filename.replace(".ftml", "").strip())
-        self.load_from_xml_data(xml_content, name)
+        if verify_waypoint_data(xml_content):
+            name = os.path.basename(filename.replace(".ftml", "").strip())
+            self.load_from_xml_data(xml_content, name)
```

Comment:
This does the verification twice, once here and once in `load_from_xml_data`.

Reply:
I'd argue that the verification only belongs into the top-level `load_from_xml_data` function here: https://github.com/Open-MSS/MSS/blob/ff744f5f4dadf09407e81688a0da18b1cd771d3e/mslib/msui/flighttrack.py#L98